tgui/packages/tgui/interfaces/common/temperatureUtils.js

```diff
@@ -0,0 +1,37 @@
+import { Color } from 'common/color';
+
+export const TemperatureColors = {
+  cold: new Color(66, 194, 255),
+  neutral: new Color(170, 170, 170),
+  hot: new Color(255, 120, 0),
+  veryhot: new Color(255, 0, 0),
+};
+
+const neutralTemperature = 293.15;
+const deviation = 200;
+const highTemperature = neutralTemperature + deviation;
+
+export const getTemperatureColor = (temperature) => {
+  const { cold, neutral, hot, veryhot } = TemperatureColors;
+
+  if (temperature < highTemperature) {
+    return Color.lookup((temperature - neutralTemperature) / (deviation * 2) + 0.5, [cold, neutral, hot]);
+  }
+  return Color.lookup((temperature - highTemperature) / (1000 - highTemperature), [hot, veryhot]);
+};
+
+export const getTemperatureIcon = (temperature) => {
+  switch (Math.round(temperature/200)) {
```

Comment:
Should this be deviation instead of 200

Reply:
`deviation` is how much the input temperature must deviate from the neutral temperature in order to fully change from the neutral color (grey) to the cold (blue) or hot (orange) color. Like, the length of the gradients between the colors.

`getTemperatureIcon` isn't based on the neutral temperature, so it doesn't specifically use deviation.